pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -945,12 +960,38 @@ public void completeHandleFuture() {
     }
 
     public void exceptionHandleFuture(Throwable t) {
-        final boolean completedNow = this.pendingAckHandleCompletableFuture.completeExceptionally(t);
+        if (isRetryableException(t)) {
+            long retryTime = backoff.next();
```

Comment:
Do we need to add a `failedRecoverTimesThreshold` for the retry? If recover times exceeds the threshold, skip retry and unload the subscription, to prevent potential infinity retry.

Reply:
In fact, `skip retry and unload the subscription` is also a retry operation. And it will reset the backoff.
There is not a better way to handle the unretryable exception. If the pendingack keeps failing, we should expect the client to throw an exception and have maintenance personnel intervene manually.